addons/binding/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/handler/EnergenieSubdevicesHandler.java

```diff
@@ -0,0 +1,618 @@
+        for (JsonSubdevice device : newDevices) {
+            EnergenieDeviceTypes type = device.getType();
+            int deviceID = device.getID();
+            int gateway = device.getParentID();
+
+            boolean isNewDeviceFound = (type == deviceType) && (gateway == gatewayID) && (!oldIDs.contains(deviceID));
+            if (isNewDeviceFound) {
+                return device;
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        ThingStatus status = getThing().getStatus();
+        if (status == ThingStatus.ONLINE) {
+            JsonSubdevice subdevice = gatewayHandler.getSubdeviceData(energenieID);
+            if (subdevice != null) {
+                if (command instanceof RefreshType && channelUID.getId().equals(CHANNEL_STATE)) {
+                    updateThingState(subdevice, channelUID);
+                } else {
+                    logger.warn("Unsupported command {} for channel with UID {}", command.toFullString(), channelUID);
+                }
+            } else {
+                logger.warn("Can't execute command {}. No data for device with ID {}", command, this.energenieID);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Subdevice " + energenieID + " is missing, it might be deleted!");
+            }
+        } else {
+            logger.warn("Device with ID {} is in status {}. Not able to handle command {} for channel with UID {}",
+                    this.energenieID, status, command.toFullString(), channelUID);
+        }
+
+    }
+
+    private void updateThingState(JsonSubdevice subdevice, ChannelUID channelUID) {
+        ThingTypeUID thingTypeUID = getThing().getThingTypeUID();
+
+        String channelID = channelUID.getId();
+        State state = null;
+
+        switch (thingTypeUID.getId()) {
+            case EnergenieBindingConstants.THING_ID_MOTION_SENSOR:
+                state = getMotionSensorState(subdevice, channelID);
+                break;
+            case EnergenieBindingConstants.THING_ID_OPEN_SENSOR:
+                state = getOpenSensorState(subdevice, channelID);
+                break;
+            case EnergenieBindingConstants.THING_ID_ENERGY_MONITOR:
+                state = getEnergyMonitorState(subdevice, channelID);
+                break;
+            default:
+                logger.warn("Channel with UID {} won't be updated! It is not supported from thing type {}", channelUID,
+                        thingTypeUID);
+                return;
+        }
+
+        if (state != null) {
+            logger.debug("About to update state for channel {} to {}", channelUID, state.toFullString());
+            updateState(channelUID, state);
+        } else {
+            logger.info("Channel with UID {} won't be updated. Server response is incomplete.", channelUID);
```

Comment:
reduce to debug.
If this situation is due to a communcation error, set the Thing status to OFFLINE.
If it can only happen in case of a software bug, log warn or error including stacktrace.